src/jaxsim/parsers/rod/parser.py

```diff
@@ -326,6 +326,18 @@ def extract_model_data(
 
                 collisions.append(sphere_collision)
 
+            if (
+                collision.geometry.mesh is not None
+                and collision.geometry.mesh.uri is not None
+            ):
+                mesh_collision = utils.create_mesh_collision(
+                    collision=collision,
+                    link_description=links_dict[link.name],
+                    mesh_uri=collision.geometry.mesh.uri,
```

Comment:
This is already part of the `collision` argument, can you extract it and use it directly inside `create_mesh_collision`? I would also remove the check if the uri is None inside this function, and raise an exception if it is. I don't think we want to support any mesh without a defined uri, and failing silently can lead to user surprise.